go-controller/pkg/factory/handler.go

```diff
@@ -160,28 +161,35 @@ func (i *informer) processEvents(events chan *event, stopChan <-chan struct{}) {
 	}
 }
 
-func getQueueNum(oType reflect.Type, obj interface{}, numEventQueues int) uint32 {
+func (i *informer) getQueueNum(oType reflect.Type, obj interface{}, numEventQueues uint32, del bool) uint32 {
+
 	meta, err := getObjectMeta(oType, obj)
 	if err != nil {
 		klog.Errorf("Object has no meta: %v", err)
 		return 0
 	}
 
-	// Distribute the object to an event queue based on a hash of its
-	// namespaced name, so that all events for a given object are
-	// serialized in one queue.
-	h := fnv.New32()
-	if meta.Namespace != "" {
-		_, _ = h.Write([]byte(meta.Namespace))
-		_, _ = h.Write([]byte("/"))
+	if del {
+		if queue, exists := i.queueMap.LoadAndDelete(meta.UID); exists {
```

Comment:
@trozet so we can't use UID as the key; we have to use namespace/name. Otherwise we get overlapping addLogicalPort/deleteLogicalPort calls for quick create/delete pods like the StatefulSet ones. Found that in the OpenShift CI. I'm going to change back to namespace + name; I think it's fine.

Reply:
Oh wow, nice catch. Yeah lgtm